db/models.py

```diff
@@ -685,3 +685,29 @@ def __repr__(self):
         return "Feedback(name='{0}', email='{1}', topic='{2}', comment='{3}')".format(
             self.name, self.email, self.topic, self.comment
         )
+
+
+class QueryData(Base):
+
+    __tablename__ = "query_data"
+
+    __table_args__ = (PrimaryKeyConstraint("client_id", "key", name="querydata_pkey"),)
+
+    client_id = Column(String, nullable=False, primary_key=True)
+
+    # Key to distinguish between different types of json data that can be stored
+    key = Column(String, nullable=False, primary_key=True)
+
```

Comment:
Add a module column before the key column

`module = Column(String, nullable=False, primary_key=True)`

Reply:
...and there should be a max column size for these string columns, probably 64 characters

Reply:
I disagree. Unnecessary paranoid namespace separation. There's no reason to associate any particular key with any particular query module, a key value pair such as username might be of value to multiple modules. Why do we need a qmodule name to disambiguate?

Reply:
Yes, valid point. Although there might then be a convention for a global namespace, such as module="global". But there could alternatively be a convention for modules to prefix their keys with their name, such as "homecontrol:device" or something like that. - The concern arises because in the future, we may have independent third parties developing modules, and it is not reasonable to expect them to know about each others' key namespaces.